### PR TITLE
Several bug fixes, support mixins and auto-bind

### PR DESCRIPTION
**Pull request by rictic**
Mixins have come up a couple times so I moved them to the head of the list.

Next I'm gonna start going through and upgrade some of my old apps that I took responsibility for upgrading.

When this PR lands I'll tag v0.1.1 and push to npm.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,10 +105,11 @@ Please file bugs as you see them! See [CONTRIBUTING.md](CONTRIBUTING.md) for mor
         version.
   - [ ] Adds additional imports when necessary.
   - [ ] Warns on other major breaking changes to elements.
-  - [ ] Upgrades mixins from 0.5 into behaviors of 1.0.
+  - [x] Upgrades mixins from 0.5 into behaviors of 1.0.
```
Comment (justinfagnani) [inline README.md]: could use some details here - does this upgrade the mixin application, declaration, or both?

File: lib/upgrade_html.js [hunk 2]
```diff
@@ -74,7 +81,7 @@ function upgradeHtml(filename) {
     domModule.text('\n');
 
     if ('extends' in polyElem.attribs) {
-      domModule.attr('extends', polyElem.attribs.extends);
+      newDeclarations.push({type: 'Property', key: {type: 'Identifier', name: 'extends'}, value: {type: 'Literal', value: polyElem.attribs.extends}});
```
Comment (justinfagnani) [inline lib/upgrade_html.js]: long line (> 80 cols)

File: README.md (hunk 1 above)
Comment (rictic) [inline README.md]: It's hopefully a bit clearer if you expand the next few lines. This is part of upgrading official elements. So upgrading references of say Polymer.CoreResizable to Polymer.IronResizableBehavior in a mixin context.

File: lib/upgrade_html.js (hunk 2 above)
Comment (rictic) [inline lib/upgrade_html.js]: Done.

File: lib/upgrade_html.js [hunk 3]
```diff
@@ -392,6 +274,186 @@ function upgradeScriptElement($, docFilename, scriptElem, attrs, hostAttrs,
...
+ *
+ * @param {jQuery} $ The cheerio jQuery object for the document as a whole.
+ * @param {TemplateElement} template The template element to upgrade.
+ * @param {?Array<Object>} opt_newDeclarations An array to put new function
+ *     declarations into. If not given, then it assumes that we're in an auto-
+ *     binding template, and adds HTML comments warning if any expressions
+ *     require functionality moved into a computed expression declaration.
+ */
+function upgradeDataBoundTemplate($, template, opt_newDeclarations) {
+  var newDeclarations = opt_newDeclarations;
```
Comment (justinfagnani) [inline lib/upgrade_html.js]: is this just a pure refactor, or did anything inside this chunk change?

Comment (rictic) [inline lib/upgrade_html.js]: Only in the three or so places that touch newDeclarations, everything else is the same.

File: README.md (hunk 1 above)
Comment (justinfagnani) [inline README.md]: it's still unclear to me from the list whether you upgrade a user-written mixin to a behavior or not.

File: lib/upgrade_html.js [hunk 4]
```diff
@@ -116,144 +127,15 @@ function upgradeHtml(filename) {
...
-      }
-      inputElem.attribs.value = '{{' + match[1] + '::input}}';
-    });
+    upgradeDataBoundTemplate($, template, newDeclarations);
 
     // Upgrade the js
     $('script', polyElem).each(function(_, scriptElem) {
       // Move the script after the polymer-element.
       $(polyElem).after(scriptElem);
+      if (upgradedScriptElems.has(scriptElem)) {
```
Comment (justinfagnani) [inline lib/upgrade_html.js]: when is this case hit? Seems like these templates should not have been selected by previous transforms.

File: README.md (hunk 1 above)
Comment (rictic) [inline README.md]: Oh I think I get what you're saying. I tried to clarify the language some more, PTAL

File: lib/upgrade_html.js (hunk 4 above)
Comment (rictic) [inline lib/upgrade_html.js]: Oh, you're right. I was trying to prevent upgrading webcomponents.js, but that needed to happen below. Removed this check.

**Comment by justinfagnani** [comment]
I see you're making a few small changes still, but so far and based on the fixtures, this looks good. Merge away when you're ready.

**Comment by rictic** [comment]
Cool, that should do it. Thanks for the quick review!

**Pull request merged**